src/top.rs

```diff
@@ -130,6 +150,27 @@ impl<'a> Iterator for LinkIter<'a> {
     }
 }
 
+#[derive(Debug, Copy, Clone, Hash, Eq, PartialEq, Ord, PartialOrd)]
+struct Target(IpAddr, Interface);
+
+#[derive(Debug, Copy, Clone, Hash, Eq, PartialEq, Ord, PartialOrd)]
+enum Interface {
+    Localhost,
+    External,
+}
+
+impl Target {
+    fn new(src: IpAddr, dst: IpAddr) -> Target {
+        if src.is_loopback() {
+            Target(dst, Interface::Localhost)
+        } else if dst.is_loopback() {
+            Target(src, Interface::Localhost)
+        } else {
+            Target(dst, Interface::External)
+        }
```

Comment:
Would be nice to have some docs explaining the logic here

Reply:
Wonderful feedback. I just realized we don't need this type.